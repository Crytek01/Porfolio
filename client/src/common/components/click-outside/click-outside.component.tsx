import React, { FunctionComponent, useEffect, useRef } from 'react';

interface IClickOutside {
  onClickOutside: () => void;
  children: React.ReactNode;
}

export const ClickOutside: FunctionComponent<IClickOutside> = ({
  onClickOutside,
  children,
}: IClickOutside) => {
  const elementRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const handleOutsideClickEvent = (event: MouseEvent) => {
      const currentElement = elementRef.current;

      if (currentElement && !currentElement.contains(event.target as Node)) {
        onClickOutside && onClickOutside();
      }
    };

    document.addEventListener('click', handleOutsideClickEvent, true);

    return () => {
      document.removeEventListener('click', handleOutsideClickEvent, true);
    };
  }, [onClickOutside]);

  return <div ref={elementRef}>{children}</div>;
};
