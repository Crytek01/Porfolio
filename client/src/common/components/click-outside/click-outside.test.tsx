import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';

import { ClickOutside } from './click-outside.component';

describe('Click outside component Test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ClickOutside
          onClickOutside={() => {
            console.log();
          }}
        >
          <div></div>
        </ClickOutside>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
