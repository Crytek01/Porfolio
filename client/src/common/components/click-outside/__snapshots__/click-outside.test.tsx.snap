// Vitest Snapshot v1

exports[`Click outside component Test > renders correctly 1`] = `
<div>
  <div />
</div>
`;
